spec/formatting.md

```diff
@@ -19,6 +19,112 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+## Pattern Selection
+
+When formatting a message with one or more selectors,
+the Pattern of one of Variants must be selected for formatting.
+
+An implementation may use any pattern selection method,
+as long as its observable behaviour matches the results of the method defined here.
+
+### Setup
+
+To select a Pattern,
+first the resolved values _res_ of the Selector must be determined:
+
+1. Let _res_ be a new empty list of resolved values that support selection.
+2. For each Expression _exp_ of the message's Selector Expressions,
+   1. Let _rv_ be the resolved value of _exp_.
+   2. If selection is supported for _rv_:
+      1. Append _rv_ as the last element of the list _res_.
+   3. Else:
+      1. Emit a Selection Error.
+      2. Let _nomatch_ be a resolved value for which selection always fails.
+      3. Append _nomatch_ as the last element of the list _res_.
+
+The shape of the resolved values must be determined by each implementation,
+along with the manner of determining their support for selection.
+
+### Variant Tests
+
+Using _res_,
+the Variants are iterated in source order and the following test is performed
+to find one with all of its keys matching the Selector Expressions:
+
+1. For each index _i_ in _res_:
+   1. Let _key_ be the VariantKey at position _i_.
+   2. If _key_ is not equal to the catch-all key `'*'`:
+      1. Let _sel_ be the entry in _res_ at position _i_.
+      2. Let _pass_ be the boolean result of testing _key_ against _sel_.
+      3. If _pass_ is False,
+         1. Return False to indicate failure.
+2. Return True to indicate success.
+
+The manner of testing _key_ against _sel_ must be defined by each implementation.
+
+Once a Variant is found for which the test succeeds,
+its Pattern is used for formatting the message.
+As each valid message must have a Variant with keys all equal to the catch-all key,
+this selection will always succeed.
+Variants after one with all catch-all keys will never be selected.
+
+### Examples
```

Comment:
We should have an example with a plural/number formatting+selection to address the comments regarding implicit assumptions about matching.

Reply:
More specifically, the intention here is that when we include an example of a plurals selection message, we have a real use case where matching is not as simple as just string comparison. We should be able run it through the above described algorithm verbiage and verify that everything we need is accounted for in the text.